app/assets/v2/js/pages/new_bounty.js

```diff
@@ -122,139 +125,120 @@ $(document).ready(function(){
-                _alert({ title: "Transaction #1 Submitted", message: msg  }, 'info');
-                next();
+                _alert({ message: "There was an error.  Please try again or contact support." });
+                $('#submitBounty').removeAttr('disabled');
+                return;
             }
-        };
-        var sendERC20Approve = function(){
-            token_contract.approve.estimateGas(bounty_address()
-                ,amount, 
-                function(errors,result){
-                    mixpanel.track("New Bounty Error", {step: 'erc20', error: errors});
-                    var gas = Math.round(erc20_approve_gas * gasMultiplier);
-                    var gasLimit = Math.round(gas * gasLimitMultiplier);
-                    token_contract.approve.sendTransaction(bounty_address()
-                        ,amount, 
-                        {from :account, 
-                            gas:web3.toHex(gas), 
-                            gasLimit: web3.toHex(gasLimit), 
-                            gasPrice:web3.toHex(defaultGasPrice), 
-                        },
-                        erc20_approve_callback);
-                });
-        };
-        // actually send the transactions to web3
-        setTimeout(function(){
-            bounty.bountydetails.call(issueURL, function(error, result){
-                if(error){
-                    console.error(error);
-                    mixpanel.track("New Bounty Error", {step: 'details', error: error});
-                    _alert({ message: "There was an error.  Please try again or contact support." });
-                    unloading_button($('#submitBounty'));
-                    return;
-                }
-                var isOpenAlready = result[4];
-                if(isOpenAlready){
-                    _alert("There is already an open funding on this issue.  Please try again with another issue.");
-                    unloading_button($('#submitBounty'));
-                    return;
-                }
-                if(!isETH){
-                    sendERC20Approve();
-                } else {
-                    sendPostBounty();
-                }
-            });
-        },100);
-        e.preventDefault();
-    });
+            localStorage['dataHash'] = result;  // cache data hash to find bountyId later
+            // bounty is a web3.js eth.contract address
+            // The Ethereum network requires using ether to do stuff on it
+            // issueAndActivateBounty is a method definied in the StandardBounties solidity contract.
+            var bountyIndex = bounty.issueAndActivateBounty(
+                account, 
+                expire_date, 
+                result, 
+                amount, 
+                0x0, 
+                false, 
+                tokenAddress,
+                amount,  // TODO:  Why is amount added twice?
```

Comment:
@mbeylin  would know

Reply:
Some of these comments was just me thinking out loud, it kind of helped me understand the code by going through and adding comments.  I'm happy to delete the comments though.  Some are useful to keep, but the ones where I ask myself questions we can prob delete.

Reply:
AH! So, the first `amount` you're using is the `fulfillmentAmount`: the amount that an individual gets paid upon successful completion of the bounty. The second `amount` is the value you're sending to the contract, which sometimes may be the same as the `fulfillmentAmount`, but may be larger (for example if you want to pay the bounty of $5 out to the top 3 submissions, you would have the `fulfillmentAmount` be $5, and the `value` be $15).